src/js/components/StarRating/StarRating.js

```diff
@@ -1,24 +1,30 @@
-import React, { useContext, useState } from 'react';
+import React, { useContext } from 'react';
 import { ThemeContext } from 'styled-components';
 import { Star } from 'grommet-icons/icons/Star';
 import { StarOutline } from 'grommet-icons/icons/StarOutline';
+import { FormContext } from '../Form/FormContext';
 import { RadioButtonGroup } from '../RadioButtonGroup';
 
-const StarRating = ({ ...rest }) => {
+const StarRating = ({ name, defaultValue, value: valueProp, ...rest }) => {
+  const formContext = useContext(FormContext);
   const theme = useContext(ThemeContext);
-  const [value, setValue] = useState();
+  const [value, setValue] = formContext.useFormInput({
+    name,
+    value: valueProp,
+    initialValue: defaultValue ?? 0,
```

Comment:
Could it just be `initialValue: defaultValue`?

Reply:
i think so, so update as requested

Reply:
I'm not sure if it's related to this change, but the StarRating storybook example isn't work anymore. I can't see any changes as I click the stars: https://deploy-preview-7077--sad-tereshkova-173d07.netlify.app/?path=/story/input-starrating-simple--simple

Reply:
Yes, it's a related issue. In this case, the initial value should be set to 0; otherwise, useFormInput won't be able to update its state if the initial value is undefined. In earlier tests, it worked because the default value was set to 0.
<img width="626" alt="Screenshot 2024-01-19 at 9 12 29 AM" src="https://github.com/grommet/grommet/assets/134603758/de0e3f56-f717-4a61-a04f-71d00681f0ef">

so it revered back to our old state 
`initialValue: defaultValue ?? 0`